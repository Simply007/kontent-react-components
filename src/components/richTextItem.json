{
  "item": {
    "system": {
      "id": "e72461f6-49d6-4308-8250-0b6e36cec15d",
      "name": "Ondřej Chrastina",
      "codename": "ondrej_chrastina",
      "language": "default",
      "type": "person",
      "collection": "default",
      "sitemap_locations": [],
      "last_modified": "2022-02-14T09:33:27.4528882Z",
      "workflow_step": "published"
    },
    "elements": {
      "name": {
        "type": "text",
        "name": "Name",
        "value": "Ondřej Chrastina"
      },
      "bio": {
        "type": "rich_text",
        "name": "Bio",
        "images": {
          "d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88": {
            "image_id": "d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88",
            "description": null,
            "url": "https://assets-us-01.kc-usercontent.com:443/0f6785d7-8214-000f-62b3-d9580d692b59/44299668-b37b-4224-a115-1fd66f7d7b36/Yprofile.jpg",
            "width": 500,
            "height": 500
          }
        },
        "links": {
          "2b805947-7ca5-4e6a-baa5-734a91f3cfa2": {
            "codename": "john_doe",
            "type": "person",
            "url_slug": "john-doe"
          }
        },
        "modular_content": [
          "ondrej_chrastina_tech",
          "simply007_kontent_gatsby_benchmark",
          "n5b76e971_4709_01f2_e8ea_36a9716d7b42"
        ],
        "value": "<p>This is Ondřej Chrastina - Developer Advocate with <a href=\"https://kontent.ai\" data-new-window=\"true\" target=\"_blank\" rel=\"noopener noreferrer\">Kentico Kontent</a>.</p>\n<figure data-asset-id=\"d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88\" data-image-id=\"d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88\"><img src=\"https://assets-us-01.kc-usercontent.com:443/0f6785d7-8214-000f-62b3-d9580d692b59/44299668-b37b-4224-a115-1fd66f7d7b36/Yprofile.jpg\" data-asset-id=\"d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88\" data-image-id=\"d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88\" alt=\"\"></figure>\n<p>He likes to do web sites. This is his latest project:</p>\n<object type=\"application/kenticocloud\" data-type=\"item\" data-rel=\"link\" data-codename=\"ondrej_chrastina_tech\"></object>\n<p><br>\nHe also likes OSS. This is his latest repository:</p>\n<object type=\"application/kenticocloud\" data-type=\"item\" data-rel=\"link\" data-codename=\"simply007_kontent_gatsby_benchmark\"></object>\n<p><br>\nOn some projects, he was cooperating with <a data-item-id=\"2b805947-7ca5-4e6a-baa5-734a91f3cfa2\" href=\"\">John Doe</a>.</p>\n<p>You could take a look at their&nbsp;<a href=\"https://google.com\" title=\"sample link\">latest project</a>.</p>\n<p>Check out also this interesting repository</p>\n<object type=\"application/kenticocloud\" data-type=\"item\" data-rel=\"component\" data-codename=\"n5b76e971_4709_01f2_e8ea_36a9716d7b42\"></object>"
      },
      "slug": {
        "type": "url_slug",
        "name": "Slug",
        "value": "ondrej-chrastina"
      }
    }
  },
  "modular_content": {
    "n5b76e971_4709_01f2_e8ea_36a9716d7b42": {
      "system": {
        "id": "5b76e971-4709-01f2-e8ea-36a9716d7b42",
        "name": "5b76e971-4709-01f2-e8ea-36a9716d7b42",
        "codename": "n5b76e971_4709_01f2_e8ea_36a9716d7b42",
        "language": "default",
        "type": "repository",
        "collection": "default",
        "sitemap_locations": [],
        "last_modified": "2022-02-14T09:33:27.4528882Z"
      },
      "elements": {
        "name": {
          "type": "text",
          "name": "Name",
          "value": "Gatsby packages repository"
        },
        "slug": {
          "type": "url_slug",
          "name": "Slug",
          "value": ""
        },
        "summary": {
          "type": "text",
          "name": "Summary",
          "value": "Gatsby packages repository containing the source plugin, components package, and a bunch of Gatsby examples."
        },
        "url": {
          "type": "text",
          "name": "URL",
          "value": "https://github.com/Kentico/kontent-gatsby-packages"
        }
      }
    },
    "ondrej_chrastina_tech": {
      "system": {
        "id": "884bd3cf-98b6-438e-b50d-a593a3bad623",
        "name": "ondrej.chrastina.tech",
        "codename": "ondrej_chrastina_tech",
        "language": "default",
        "type": "website",
        "collection": "default",
        "sitemap_locations": [],
        "last_modified": "2020-08-14T13:18:13.858424Z",
        "workflow_step": "published"
      },
      "elements": {
        "name": {
          "type": "text",
          "name": "Name",
          "value": "ondrej.chrastina.tech"
        },
        "summary": {
          "type": "text",
          "name": "Summary",
          "value": "Ondřej Chrastina's personal site"
        },
        "url": {
          "type": "text",
          "name": "URL",
          "value": "https://ondrej.chrastina.tech"
        },
        "slug": {
          "type": "url_slug",
          "name": "Slug",
          "value": "ondrej-chrastina-tech"
        },
        "source_repository": {
          "type": "modular_content",
          "name": "Source repository",
          "value": [
            "simply007_simply007_github_io"
          ]
        }
      }
    },
    "simply007_kontent_gatsby_benchmark": {
      "system": {
        "id": "16ec0ca3-8525-4154-ba84-0bb7ec712bf6",
        "name": "Simply007/kontent-gatsby-benchmark",
        "codename": "simply007_kontent_gatsby_benchmark",
        "language": "default",
        "type": "repository",
        "collection": "default",
        "sitemap_locations": [],
        "last_modified": "2020-08-14T12:19:40.5439642Z",
        "workflow_step": "published"
      },
      "elements": {
        "name": {
          "type": "text",
          "name": "Name",
          "value": "Simply007/kontent-gatsby-benchmark"
        },
        "summary": {
          "type": "text",
          "name": "Summary",
          "value": "Example site for Kontent as a source for \"Will It Build\". Should/Will be generalized into e.g. a theme."
        },
        "url": {
          "type": "text",
          "name": "URL",
          "value": "https://github.com/Simply007/kontent-gatsby-benchmark"
        },
        "slug": {
          "type": "url_slug",
          "name": "Slug",
          "value": "simply007-kontent-gatsby-benchmark"
        }
      }
    }
  }
}