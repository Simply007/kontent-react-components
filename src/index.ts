export * from './components/rich-text-element';

export { isLinkedItem, isComponent } from './utils/richTextUtils';