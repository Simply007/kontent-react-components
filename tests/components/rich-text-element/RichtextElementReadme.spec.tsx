import { createDeliveryClient, Elements } from '@kentico/kontent-delivery';
import { isComponent, isLinkedItem, RichTextElement } from '../../../src';
import richTextItem from './complexRichTextItem.json';
import TestRenderer from 'react-test-renderer';
import { Element as DomHandlerElement } from 'domhandler';

describe('<RichTextElement/> readme code samples', () => {

    var mockClient = createDeliveryClient({
        projectId: 'dummyClient',
    });

    const response = mockClient.item("dummyItem")
        .map(richTextItem);

    it('should resolve without fail', () => {
        const testRenderer = TestRenderer.create(
            <RichTextElement
                richTextElement={response.item.elements["bio"] as Elements.RichTextElement}
                resolvers={{
                    resolveLinkedItem: (linkedItem, { domElement, domToReact }) => {
                        if (isComponent(domElement)) {
                            return (
                                <>
                                    <h1>Component</h1>
                                    <pre>{JSON.stringify(linkedItem, undefined, 2)}</pre>;
                                </>
                            );
                        }
                        if (isLinkedItem(domElement)) {
                            return (
                                <>
                                    <h1>Linked item</h1>
                                    <pre>{JSON.stringify(linkedItem, undefined, 2)}</pre>;
                                </>
                            );
                        }
                        throw new Error("Unknown type of the linked item's dom node");
                    },
                    resolveImage: (image, { domElement, domToReact }): JSX.Element => (
                        <img
                            src={image.url}
                            alt={image.description ? image.description : image.imageId}
                            width="200"
                        />
                    ),
                    resolveLink: (link, { domElement, domToReact }): JSX.Element => (
                        <a href={`/${link.type}/${link.urlSlug}`}>
                            {domToReact(domElement.children)}
                        </a>
                    ),
                    resolveDomNode: ({ domNode, domToReact }) => {
                        if (domNode instanceof DomHandlerElement && domNode.name === 'table') {
                            return <div className="table-wrapper">{domToReact([domNode])}</div>;
                        }
                    }
                }}
            />
        );
        expect(testRenderer.toJSON()).toMatchSnapshot();
    })
});