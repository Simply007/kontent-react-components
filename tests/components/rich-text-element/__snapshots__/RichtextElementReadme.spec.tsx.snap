// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RichTextElement/> readme code samples should resolve without fail 1`] = `
Array [
  <p>
    This is Ondřej Chrastina - Developer Advocate with 
    <a
      data-new-window="true"
      href="https://kontent.ai"
      rel="noopener noreferrer"
      target="_blank"
    >
      Kentico Kontent
    </a>
    .
  </p>,
  <img
    alt="d32b8ad5-0cf4-47a8-8b53-ed4a1e80dc88"
    src="https://assets-us-01.kc-usercontent.com:443/0f6785d7-8214-000f-62b3-d9580d692b59/44299668-b37b-4224-a115-1fd66f7d7b36/Yprofile.jpg"
    width="200"
  />,
  "
",
  <p>
    He likes to do web sites. This is his latest project:
  </p>,
  "
",
  <h1>
    Linked item
  </h1>,
  <pre>
    {
  "elements": {
    "name": {
      "name": "Name",
      "type": "text",
      "value": "ondrej.chrastina.tech"
    },
    "summary": {
      "name": "Summary",
      "type": "text",
      "value": "Ondřej Chrastina's personal site"
    },
    "url": {
      "name": "URL",
      "type": "text",
      "value": "https://ondrej.chrastina.tech"
    },
    "slug": {
      "name": "Slug",
      "type": "url_slug",
      "value": "ondrej-chrastina-tech"
    },
    "source_repository": {
      "name": "Source repository",
      "type": "modular_content",
      "value": [
        "simply007_simply007_github_io"
      ],
      "linkedItems": []
    }
  },
  "system": {
    "codename": "ondrej_chrastina_tech",
    "collection": "default",
    "id": "884bd3cf-98b6-438e-b50d-a593a3bad623",
    "language": "default",
    "lastModified": "2020-08-14T13:18:13.858424Z",
    "name": "ondrej.chrastina.tech",
    "sitemapLocations": [],
    "type": "website",
    "workflowStep": "published"
  }
}
  </pre>,
  ";",
  "
",
  <p>
    <br />
    
He also likes OSS. This is his latest repository:
  </p>,
  "
",
  <h1>
    Linked item
  </h1>,
  <pre>
    {
  "elements": {
    "name": {
      "name": "Name",
      "type": "text",
      "value": "Simply007/kontent-gatsby-benchmark"
    },
    "summary": {
      "name": "Summary",
      "type": "text",
      "value": "Example site for Kontent as a source for \\"Will It Build\\". Should/Will be generalized into e.g. a theme."
    },
    "url": {
      "name": "URL",
      "type": "text",
      "value": "https://github.com/Simply007/kontent-gatsby-benchmark"
    },
    "slug": {
      "name": "Slug",
      "type": "url_slug",
      "value": "simply007-kontent-gatsby-benchmark"
    }
  },
  "system": {
    "codename": "simply007_kontent_gatsby_benchmark",
    "collection": "default",
    "id": "16ec0ca3-8525-4154-ba84-0bb7ec712bf6",
    "language": "default",
    "lastModified": "2020-08-14T12:19:40.5439642Z",
    "name": "Simply007/kontent-gatsby-benchmark",
    "sitemapLocations": [],
    "type": "repository",
    "workflowStep": "published"
  }
}
  </pre>,
  ";",
  "
",
  <p>
    <br />
    
On some projects, he was cooperating with 
    <a
      href="/person/john-doe"
    >
      John Doe
    </a>
    .
  </p>,
  "
",
  <p>
    You could take a look at their 
    <a
      href="https://google.com"
      title="sample link"
    >
      latest project
    </a>
    .
  </p>,
  "
",
  <p>
    Check out also this interesting repository
  </p>,
  "
",
  <h1>
    Component
  </h1>,
  <pre>
    {
  "elements": {
    "name": {
      "name": "Name",
      "type": "text",
      "value": "Gatsby packages repository"
    },
    "slug": {
      "name": "Slug",
      "type": "url_slug",
      "value": ""
    },
    "summary": {
      "name": "Summary",
      "type": "text",
      "value": "Gatsby packages repository containing the source plugin, components package, and a bunch of Gatsby examples."
    },
    "url": {
      "name": "URL",
      "type": "text",
      "value": "https://github.com/Kentico/kontent-gatsby-packages"
    }
  },
  "system": {
    "codename": "n5b76e971_4709_01f2_e8ea_36a9716d7b42",
    "collection": "default",
    "id": "5b76e971-4709-01f2-e8ea-36a9716d7b42",
    "language": "default",
    "lastModified": "2022-02-14T09:33:27.4528882Z",
    "name": "5b76e971-4709-01f2-e8ea-36a9716d7b42",
    "sitemapLocations": [],
    "type": "repository",
    "workflowStep": null
  }
}
  </pre>,
  ";",
]
`;
